### PR TITLE
Simnode tests in Nix

### PR DESCRIPTION
**Pull request by hussein-aitlahcen**
Fixup for https://github.com/ComposableFi/composable/pull/1450#issuecomment-1225368528

### CODE REVIEW

File: .github/workflows/build-test-deploy.yml [hunk 1]
```diff
@@ -187,6 +187,37 @@ jobs:
...
+  package-simnode-tests:
+    name: "Package Simnode Tests"
+    needs: common-deps
+    runs-on:
+      - self-hosted
+      - linux
+      - x64
+      - sre
+    concurrency:
+      group: packagecomposablenode-${{ github.ref }}
```
Comment (dzmitry-lahoda) [inline .github/workflows/build-test-deploy.yml:199]: package-simnode-tests-${{ github.ref }}

**Review by dzmitry-lahoda — APPROVED** [review]

File: .github/workflows/build-test-deploy.yml (hunk 1 above)
Comment (hussein-aitlahcen) [reply .github/workflows/build-test-deploy.yml]: :+1:

**Comment by hussein-aitlahcen** [comment]
@seunlanlege I've moved simnode as a nix job, the missing part now is the DB, I'm syncing with @haroldsphinx on that. Will keep you updated.

File: .github/workflows/build-test-deploy.yml [hunk 2]
```diff
@@ -206,7 +206,46 @@ jobs:
...
+      matrix:
+        runtime: [picasso, composable]
+    container:
+      image: niteo/nixpkgs-nixos-22.05:316b762afdb9e142a803f29c49a88b4a47db80ee
+    steps:
+      - uses: actions/checkout@v3
+      - run: |
+          echo "experimental-features = nix-command flakes" > /etc/nix/nix.conf
+          echo "sandbox = relaxed" >> /etc/nix/nix.conf
+          echo "narinfo-cache-negative-ttl = 0" >> /etc/nix/nix.conf
```
Comment (cor) [inline .github/workflows/build-test-deploy.yml:232]: Can we combine these into a single `echo` statement? or have a file that we `cp` to it

File: .github/workflows/build-test-deploy.yml [hunk 3]
```diff
@@ -206,7 +206,46 @@ jobs:
...
+      - uses: cachix/cachix-action@f5f67badd061acb62b5c6e25e763572ca8317004
+        with:
+          skipPush: true
+          installCommand: |
+            nix-channel --add https://nixos.org/channels/nixpkgs-22.05-darwin nixpkgs
```
Comment (cor) [inline .github/workflows/build-test-deploy.yml:237]: why do we need darwin for simnode?

File: .github/workflows/build-test-deploy.yml [hunk 4]
```diff
@@ -239,7 +278,37 @@ jobs:
...
+      group: packagesimnodetests-${{ matrix.runtime }}-${{ github.ref }}
+      cancel-in-progress: true
+    container:
+      image: niteo/nixpkgs-nixos-22.05:316b762afdb9e142a803f29c49a88b4a47db80ee
+    steps:
+      - uses: actions/checkout@v3
+      - run: |
+          echo "experimental-features = nix-command flakes" > /etc/nix/nix.conf
+          echo "sandbox = relaxed" >> /etc/nix/nix.conf
+          echo "narinfo-cache-negative-ttl = 0" >> /etc/nix/nix.conf
```
Comment (cor) [inline .github/workflows/build-test-deploy.yml:300]: I think we should have a single nix.conf that we `cp` every time to make this simpler and reduce repitition

File: .github/workflows/build-test-deploy.yml [hunk 5]
```diff
@@ -239,7 +278,37 @@ jobs:
...
+      - uses: cachix/cachix-action@f5f67badd061acb62b5c6e25e763572ca8317004
+        with:
+          skipPush: true
+          installCommand: |
+            nix-channel --add https://nixos.org/channels/nixpkgs-22.05-darwin nixpkgs
```
Comment (cor) [inline .github/workflows/build-test-deploy.yml:305]: why do we need darwin here?

**Review by cor — CHANGES_REQUESTED** [review]
lgtm but some small changes requested

File: .github/workflows/build-test-deploy.yml (hunk 2 above)
Comment (hussein-aitlahcen) [reply .github/workflows/build-test-deploy.yml:232]: Let's keep that for another cleanup PR it's not the purpose of this one

File: .github/workflows/build-test-deploy.yml (hunk 3 above)
Comment (hussein-aitlahcen) [reply .github/workflows/build-test-deploy.yml:237]: There is no `nixpkgs` channels named without darwin.

**Review by KaiserKarel — APPROVED** [review]

**Comment by KaiserKarel** [comment]
Small question @hussein-aitlahcen. Is the simnode run cached based on the runtime + blocks imported?

**Comment by hussein-aitlahcen** [comment]
> Small question @hussein-aitlahcen. Is the simnode run cached based on the runtime + blocks imported?

Currently I won't cache it but we can cache it via a simple derivation: taking the hash of the DB against which we ran

File: .github/workflows/build-test-deploy.yml [hunk 6]
```diff
@@ -398,6 +437,36 @@ jobs:
       - run: |
           cachix watch-exec -j 16 -c 0 composable-community nix -- build .#check-${{ matrix.runtime }}-integration-tests --no-update-lock-file --show-trace -L       
 
+  package-simnode-tests:
```
Comment (dzmitry-lahoda) [inline .github/workflows/build-test-deploy.yml:440]: just comment, so we can follow terminology of nix, nixops. build, check, deploy.  so package is build.

Comment (dzmitry-lahoda) [reply .github/workflows/build-test-deploy.yml:440]: not for now. just note.

**Review by dzmitry-lahoda — APPROVED** [review]

File: runtime/picasso/Cargo.toml [hunk 7]
```diff
@@ -123,7 +123,7 @@ orml-xcm-support = { git = "https://github.com/open-web3-stack/open-runtime-modu
 orml-xtokens = { git = "https://github.com/open-web3-stack/open-runtime-module-library", rev = "377213f750755cc48e80a3131eaae63b5eda8362", default-features = false }
 
 # simnode support
-simnode-apis = { package = "simnode-runtime-apis", git = "https://github.com/polytope-labs/substrate-simnode", default-features = false, rev = "eed85119358bc80bbb1aedd98778f370d4932dee" }
+simnode-apis = { package = "simnode-runtime-apis", git = "https://github.com/polytope-labs/substrate-simnode", default-features = false, branch = "polkadot-v0.9.27" }
```
Comment (dzmitry-lahoda) [inline runtime/picasso/Cargo.toml:126]: not sure, hope that will not be issue https://github.com/polytope-labs/substrate-simnode/issues/15

File: .github/workflows/build-test-deploy.yml (hunk 6 above)
Comment (hussein-aitlahcen) [reply .github/workflows/build-test-deploy.yml:440]: it's a binary

**Review by dzmitry-lahoda — APPROVED** [review]

**Review by seunlanlege — APPROVED** [review]

**Review by cor — APPROVED** [review]
LGTM

**Pull request merged**